types.go

```diff
@@ -1,16 +1,114 @@
 package dmetering
 
-// Event represents a metering event that needs to be recorded
-type Event struct {
-	// Defines the service that emitted the event (firehose, substreams, evmx ...)
-	Service string `json:"service"`
-	// Defines the method within the service  (grpc_blocks, grpc_run, http_get_state, ....)
-	Method         string            `json:"method"`
-	Metadata       map[string]string `json:"metadata"`
-	RequestsCount  uint64            `json:"requests_count,omitempty"`
-	ResponsesCount uint64            `json:"responses_count,omitempty"`
-	IngressBytes   float64           `json:"ingress_bytes,omitempty"`
-	EgressBytes    float64           `json:"egress_bytes,omitempty"`
-	WrittenBytes   float64           `json:"written_bytes,omitempty"`
-	ReadBytes      float64           `json:"read_bytes,omitempty"`
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	pbmetering "github.com/streamingfast/dmetering/pb/sf/metering/v1"
+
+	"github.com/streamingfast/dgrpc"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+func init() {
```

Comment:
We should really refrain from using `init`. I much prefer to expose a `RegisterGRPC` from the interface and do the `func init() { dmetering.RegisterGRPC }` than implicit `init`.

Reply:
What about this one? Maybe we discuss together later.